src/drt/src/io/io_parser_helper.cpp

```diff
@@ -219,6 +219,135 @@ void io::Parser::initDefaultVias()
   }
 }
 
+namespace {
+std::pair<frCoord, frCoord> getBloatingDist(frTechObject* tech,
+                                            frVia via,
+                                            bool above)
+{
+  auto cut_layer = tech->getLayer(via.getViaDef()->getCutLayerNum());
+  auto enc_box = above ? via.getLayer2BBox() : via.getLayer1BBox();
+  auto cut_box = via.getCutBBox();
+  frCoord horz_overhang = 0;
+  frCoord vert_overhang = 0;
+  horz_overhang = std::min(enc_box.xMax() - cut_box.xMax(),
+                           cut_box.xMin() - enc_box.xMin());
+  vert_overhang = std::min(enc_box.yMax() - cut_box.yMax(),
+                           cut_box.yMin() - enc_box.yMin());
+  bool eol_is_horz = enc_box.dx() < enc_box.dy();
+  for (auto con : cut_layer->getLef58EnclosureConstraints(
+           via.getViaDef()->getCutClassIdx(), 0, above, true)) {
+    if (!con->isEolOnly()) {
+      break;
+    }
+    if ((eol_is_horz ? enc_box.dx() : enc_box.dy()) > con->getEolLength()) {
+      continue;
+    }
+    std::pair<frCoord, frCoord> bloats;
+    if (eol_is_horz) {
+      bloats = {std::max(0, con->getFirstOverhang() - vert_overhang),
+                std::max(0, con->getSecondOverhang() - horz_overhang)};
+    } else {
+      bloats = {std::max(0, con->getSecondOverhang() - vert_overhang),
+                std::max(0, con->getFirstOverhang() - horz_overhang)};
+    }
+    return bloats;
+  }
+  return std::make_pair<frCoord, frCoord>(0, 0);
+}
+
+}  // namespace
+
+void io::Parser::initSecondaryVias()
```

Comment:
It would be helpful to have a descriptive comment.  It isn't clear what a 'secondary via' is.